eng/targets/Imports.targets

```diff
@@ -208,6 +208,14 @@
     <None Include="$(ThirdPartyNoticesFilePath)" PackagePath="ThirdPartyNotices.rtf" Visible="false" Pack="true"/>
   </ItemGroup>
 
+  <!--
+    Microsoft.VisualStudio.Extensibility.JsonGenerators.Sdk.props creates content items for all files in .vsextension dir.
+    NuGet includes content items in package, by default but these items should only be included in the VSIX.
+  -->
```

Comment:
Why did we need to make this change for non-VS targets? My instinct would be that instead of this change we should be including VisualStudio.targets in more places. 

@dibarbet

Reply:
The .vsextension files end up in the pack content because of this package ref: https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/CSharp/Impl/Microsoft.VisualStudio.LanguageServices.CSharp.csproj#L46

```
  <ItemGroup>
    <Reference Include="System.Design" />
    <Reference Include="System.Windows.Forms" />
    <PackageReference Include="Microsoft.VisualStudio.Extensibility" />
    <PackageReference Include="Microsoft.VisualStudio.Extensibility.JsonGenerators.Sdk" PrivateAssets="all" /> <-- THIS
  </ItemGroup>
```

That ends up importing a props file which creates the content items. I could make this conditional on vssdk usage, but it seems like a better approach to keep this here, and just not conditionalize the .vsextension Pack exclusion on `IsUsingToolVsSdk`, which was what was implicitly happening due to https://github.com/dotnet/roslyn/blob/main/eng/targets/Imports.targets#L4

```
  <Import Project="VisualStudio.targets" Condition="'$(UsingToolVSSDK)' == 'true'"/>
```

Reply:
Feel like that should be importing the `VisualStudio.targets` file. @dibarbet will know better though.

Reply:
I'm not super familiar with this stuff here tbh - I'll delegate to @tmat who added the jsongenerators reference for gladstone.

Reply:
I've been a bit out of the loop as well - why are we building VS layer projects on linux?  Seems like the packages produced for `Microsoft.VisualStudio.LanguageServices.CSharp.csproj`  and others wouldn't be useful without the vssdk?

Reply:
I think we generally build everything and it seems to work fine. It's the packing that generally doesn't work, at least in release modes.